### ISSUE TITLE
Multiplayer: Particular Levels Crash Editor Server

### ISSUE THREAD

**Issue by mcphedar** [o3de/o3de#13988]
**Describe the bug**
Multiplayer server, when launched from the editor, disconnects. Running the same level using the standalone Server and Game Launcher by hand works.

**Steps to reproduce**
1. Git pull and build the latest game branch _mcphedar/Starbase_Layout_ from https://github.com/aws-lumberyard-dev/o3de-multiplayersample.git
1. Open the editor
1. Open the level ArenaWIPAlpha
1. Hit ctrl-G to start the server and client
1. Notice that the server disconnects after a few seconds

**Expected behavior**
The Server is stable and the client connects, allowing the player to move around in the level.

**Actual behavior**
The server crashes

**Screenshots/Video**
[Server.log](https://github.com/o3de/o3de/files/10386946/Server.log)

[Editor.log](https://github.com/o3de/o3de/files/10386948/Editor.log)

**Found in Branch**
 https://github.com/aws-lumberyard-dev/o3de-multiplayersample.git
Branch:  mcphedar/Starbase_Layout

**Comment by AMZN-Gene** [comment]
Looks like a problem assigning root spawnable.

```
<15:14:14> (EditorServer) - LevelSystem: Level load complete: 'Root.spawnable'

<15:14:14> (EditorServer) - System: > LoadLevel : <empty>

<15:14:14> (EditorServer) - System: Editor Server completed receiving the editor's level assets, responding to Editor...

<15:14:14> (EditorServer) - Spawnables: New root spawnable 'Root.spawnable' assigned (generation: 1).

<15:14:14> (EditorServer) - System: 

<15:14:14> (EditorServer) - ==================================================================

<15:14:14> (EditorServer) - System: Trace::Assert

<15:14:14> (EditorServer) -  C:\O3DE-Engine\Code\Framework\AzCore\AzCore/std/containers/vector.h(542): (35736) 'class std::unique_ptr<class AZ::Entity,struct std::default_delete<class AZ::Entity> > &__cdecl AZStd::vector<class std::unique_ptr<class AZ::Entity,struct std::default_delete<class AZ::Entity> >,class AZStd::allocator>::operator [](unsigned __int64)'

<15:14:14> (EditorServer) - System: AZStd::vector<>::at - position is out of range

<15:14:14> (EditorServer) - System: ------------------------------------------------

<15:14:15> (EditorServer) - System: C:\O3DE-Engine\Code\Framework\AzFramework\AzFramework\Spawnable\SpawnableEntitiesManager.cpp (373) : AzFramework::SpawnableEntitiesManager::CloneSingleAliasedEntity

<15:14:15> (EditorServer) - System: C:\O3DE-Engine\Code\Framework\AzFramework\AzFramework\Spawnable\SpawnableEntitiesManager.cpp (512) : AzFramework::SpawnableEntitiesManager::ProcessRequest

<15:14:15> (EditorServer) - System: C:\O3DE-Engine\Code\Framework\AzFramework\AzFramework\Spawnable\SpawnableEntitiesManager.cpp (273) : AzFramework::SpawnableEntitiesManager::ProcessQueue

<15:14:15> (EditorServer) - System: C:\O3DE-Engine\Code\Framework\AzFramework\AzFramework\Spawnable\SpawnableEntitiesManager.cpp (224) : AzFramework::SpawnableEntitiesManager::ProcessQueue

<15:14:15> (EditorServer) - System: C:\O3DE-Engine\Code\Framework\AzFramework\AzFramework\Spawnable\SpawnableSystemComponent.cpp (55) : AzFramework::SpawnableSystemComponent::OnTick

<15:14:15> (EditorServer) - System: C:\O3DE-Engine\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1360) : AZ::Internal::EBusContainer<AZ::TickEvents,AZ::TickEvents,0,2>::Dispatcher<AZ::EBus<AZ::TickEvents,AZ::TickEvents> >::Broadcast<void (__cdecl AZ::TickEvents::*)(float,AZ::S

<15:14:15> (EditorServer) - System: C:\O3DE-Engine\Code\Framework\AzCore\AzCore\Component\ComponentApplication.cpp (1476) : AZ::ComponentApplication::Tick

<15:14:15> (EditorServer) - System: C:\O3DE-Engine\Code\LauncherUnified\Launcher.cpp (585) : O3DELauncher::Run
```

**Comment by lmbr-pip** [comment]
Crash is in prefab system while working with network spawnables. Need help from prefab team I believe.

**Comment by lmbr-pip** [comment]
Issue maybe with the spawnable, where the spawnable maybe referencing an incorrect number of entities. Need to validate content and either fix content, or move issue to prefabs and sig/content.

**Comment by AMZN-Gene** [comment]
Spoke to spawnable team. They suggest @o3de/sig-network  look in the code when we go to spawn the entity and ensure the entity count looks right.
See _MultiplayerEditorConnection.cpp_

**Comment by mbalfour-amzn** [comment]
Some of the issues should be addressed here: https://github.com/o3de/o3de/pull/14178

**Comment by mbalfour-amzn** [comment]
One of the key errors you can see in the Editor.log is this one:
<15:14:03> [Error] (System) - Send ringbuffer full, dropped packet

This means that when the level data is getting sent from the Editor to EditorServer, one or more 20 KB from the middle of the data is simply being dropped and not sent. After that point, any number of problems can occur when trying to deserialize the data.

There are many other errors in the logs as well, so I can't guarantee how many issues are fixed by #14178 but it will at least address the "ringbuffer full" portion of it.
